### PR TITLE
feat: Add workaround for support of isAssociated

### LINKED ISSUE #2810: Add workaround for support of isAssociated
### Problem

[HIP 719](https://hips.hedera.com/hip/hip-719) adds 3 new precompile methods, of which `isAssociated()` is defined as a `view function`. Thus it should be called via `eth_call` and the Relay will call the Mirror Node Web3 module to get the result. However the EVM in the Mirror node is not updated with the latest changes related to this HIP.

### Solution

Add a temporary exception for this method only. When this method selector `0x4d8fdd6d` is detected the relay should call Consensus instead of the Mirror node.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
- Adds a temporary workaround that enables the support of direct `isAssociated` calls to the relay.
- Adds a new feature flag `ETH_CALL_FORCE_TO_CONSENSUS_BY_SELECTOR` that enables some requests to `eth_call` to go through the Consensus node instead of the Mirror node regardless of the value of `ETH_CALL_DEFAULT_TO_CONSENSUS_NODE`. When set to `true` if the function selector is found in a predetermined array then the call will go through Consensus. For now the only entry in this list is `0x4d8fdd6d`, which is the selector of `isAssociated()`

**Related issue(s)**:

Fixes #2810

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -1651,18 +1652,26 @@ export class EthImpl implements Eth {
...
+
+    const shouldForceToConsensus =
+      process.env.ETH_CALL_FORCE_TO_CONSENSUS_BY_SELECTOR == 'true' &&
+      constants.ETH_CALL_SELECTORS_ALWAYS_TO_CONSENSUS.indexOf(selector) !== -1;
+
+    // ETH_CALL_DEFAULT_TO_CONSENSUS_NODE = false enables the use of Mirror node
+    const shouldDefaultToConsensus = !(
+      process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE === undefined ||
+      process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE == 'false'
+    );
```
Comment (quiet-node) [inline packages/relay/src/lib/eth.ts:1665]: I believe this would equivalent to below
```suggestion
     const shouldDefaultToConsensus = process.env.ETH_CALL_DEFAULT_TO_CONSENSUS_NODE === 'true'
```

File: docs/configuration.md [hunk 2]
```diff
@@ -48,60 +48,61 @@ Unless you need to set a non-default value, it is recommended to only populate o
...
+| `ETH_GET_GAS_PRICE_CACHE_TTL_MS`            | "1_800_000"              | Time in ms to cache ethGasPrice returned                                                                                                                                                                                                                                                        |
+| `ETH_CALL_DEFAULT_TO_CONSENSUS_NODE`        | "false"                  | Flag to set if eth_call logic should first query the mirror node.                                                                                                                                                                                                                               |
+| `ETH_CALL_FORCE_TO_CONSENSUS_BY_SELECTOR`   | "false"                  | Flag to force eth_call logic to use consensus node for speciffic function selectors.                                                                                                                                                                                                            |
```
Comment (ebadiere) [inline docs/configuration.md:65]: Typo, "specific"

**Review by ebadiere — COMMENTED** [review]
Good work. Some comments in-line.

File: packages/relay/src/lib/eth.ts [hunk 3]
```diff
@@ -1630,12 +1631,12 @@ export class EthImpl implements Eth {
...
     const callData = call.data ? call.data : call.input;
     // log request
-    this.logger.trace(
+    this.logger.info(
       `${requestIdPrefix} call({to=${call.to}, from=${call.from}, data=${callData}, gas=${call.gas}, gasPrice=${call.gasPrice} blockParam=${blockParam}, estimate=${call.estimate})`,
     );
     // log call data size
     const callDataSize = callData ? callData.length : 0;
-    this.logger.trace(`${requestIdPrefix} call data size: ${callDataSize}`);
+    this.logger.info(`${requestIdPrefix} call data size: ${callDataSize}`);
```
[the remark this answers is not in the document]
Comment (Ivo-Yankov) [reply packages/relay/src/lib/eth.ts]: I must have forgotten to change it back after doing some debugging

**Review by ebadiere — APPROVED** [review]
LG.

File: packages/relay/tests/lib/eth/eth_call.spec.ts [hunk 4]
```diff
@@ -867,4 +867,62 @@ describe('@ethCall Eth Call spec', async function () {
...
+    let initialForceToCensensusBySelectorFF;
+    let initialEthCallConesneusFF;
+    const REDIRECTED_SELECTOR = '0x4d8fdd6d';
+    const NON_REDIRECTED_SELECTOR = '0xaaaaaaaa';
+    let callConsensusNodeSpy: sinon.SinonSpy;
+    let callMirrorNodeSpy: sinon.SinonSpy;
+    let sandbox: sinon.SinonSandbox;
+
+    before(() => {
+      initialForceToCensensusBySelectorFF = process.env.ETH_CALL_FORCE_TO_CONSENSUS_BY_SELECTOR;
```
Comment (konstantinabl) [inline packages/relay/tests/lib/eth/eth_call.spec.ts:881]: some typos in consensus here

File: packages/relay/src/formatters.ts [hunk 5]
```diff
@@ -297,6 +299,38 @@ const isHex = (value: string): boolean => {
...
 
+/**
+ * Returns the sum of all transfer amounts for the specified account. The amount is negative if the account is charged,
+ * it is positive if the account is receiving it, thus the amount is first negated and then added to the sum.
+ *
+ * @param {TransactionRecord | MirrorNodeTransactionRecord} transactionRecord - The transaction record containing transfer information.
+ * @param {string} accountId - The account ID to filter transfers.
+ * @returns {number} - The sum of transfer amounts for the specified account.
+ */
+const getTransferAmountSumForAccount = (
```
Comment (konstantinabl) [inline packages/relay/src/formatters.ts:310]: why is this added?

**Review by konstantinabl — CHANGES_REQUESTED** [review]

File: packages/relay/src/formatters.ts (hunk 5 above)
Comment (Ivo-Yankov) [reply packages/relay/src/formatters.ts]: I think this was added by mistake when I was resolving some conflicts

**Review by konstantinabl — APPROVED** [review]
LGTM

**Review by natanasow — APPROVED** [review]
